server/konnectors/orange_mobile.js

```diff
@@ -0,0 +1,219 @@
+      connector.logger.error(body);
+    }
+
+    if (err) {
+      connector.logger.error(`Download failed: ${err}`);
+      return callback(err);
+    }
+    callback(null, body);
+  });
+}
+
+function downloadGeoloc(requiredFields, entries, data, next) {
+  connector.logger.info('Downloading geoloc data from Orange...');
+
+  let uri = `${API_ROOT}/data/loc`;
+  if (requiredFields.lastGeoPoint) {
+    uri += `?start=${requiredFields.lastGeoPoint.slice(0, 19)}`;
+  }
+
+  requestOrange(uri, requiredFields.access_token, (err, body) => {
+    if (err) { return next(err); }
+    entries.geopoints = [];
+    body.forEach((point) => {
+      if (point.ts && requiredFields.lastGeoPoint < point.ts) {
+        requiredFields.lastGeoPoint = point.ts;
+      }
+      if (point.err) { return; }
+
+      entries.geopoints.push({
+        docType: 'GeoPoint',
+        docTypeVersion: connector.doctypeVersion,
+        msisdn: point.msisdn,
+        timestamp: point.ts,
+        longitude: point.loc[0],
+        latitude: point.loc[1],
+        radius: point.rad,
+      });
+    });
+
+    next();
+  });
+}
+
+function downloadCRA(requiredFields, entries, data, next) {
+  connector.logger.info('Downloading CRA data from Orange...');
+
+  let uri = `${API_ROOT}/data/cra`;
+  if (requiredFields.lastPhoneCommunicationLog) {
+    uri += `?start=${requiredFields.lastPhoneCommunicationLog.slice(0, 19)}`;
+  }
+
+  requestOrange(uri, requiredFields.access_token, (err, body) => {
+    if (err) { return next(err); }
+
+    // map SMS_C for further concat in one SMS object.
+    const smsCByTs = body.filter(cra => cra.desc.indexOf('SMS_C') === 0)
+      .reduce((agg, smsC) => {
+        agg[smsC.ts] = smsC;
+        return agg;
+      }, {});
+
+    entries.phonecommunicationlogs = [];
+
+    body.forEach((cra) => {
+      try {
+        if (cra.time && requiredFields.lastPhoneCommunicationLog < cra.time) {
+          requiredFields.lastPhoneCommunicationLog = cra.time;
+        }
+        if (cra.err || cra.desc.indexOf('SMS_C') === 0) { return; }
+
+        if (cra.desc.indexOf('SMS ') === 0) {
+          // Try to merge informations
+          const smsC = smsCByTs[cra.ts];
+          if (smsC) {
+            cra.length = smsC.units;
+            cra.chipType = 'c';
+          }
+        }
+
+        entries.phonecommunicationlogs.push({
+          docType: 'PhoneCommunicationLog',
+          docTypeVersion: connector.doctypeVersion,
+          timestamp: cra.time,
+          msisdn: cra.msisdn,
+          partner: cra.partner,
+          length: cra.units,
+          chipType: cra.typ_units,
+          longitude: cra.loc ? cra.loc[0] : undefined,
+          latitude: cra.loc ? cra.loc[1] : undefined,
+          networkType: cra.net_lbl,
+          type: cra.desc,
+          endCause: cra.end_cause,
+        });
+      } catch (e) {
+        connector.logger.error('While parsing CRA.');
+        connector.logger.error(e);
```

Comment:
I prefer not to. The aim is to have a continue on error, against some format unexpections.